### PR TITLE
tasks/build: Redo build process.

### PR DESCRIPTION
**Pull request by NodeGuy**
This is intended to fix #735, #740, and #741.

Build SDK from source every time we build Voyager because it's changing so fast.
Build all three platforms at once because it doesn't save much time to build them individually.

### CODE REVIEW

File: tasks/build/Dockerfile [hunk 1]
```diff
@@ -2,84 +2,95 @@ FROM golang:1.10.1
...
+RUN apt-get update && \
+  apt-get install --yes xz-utils
+
+# from https://raw.githubusercontent.com/nodejs/docker-node/773be66016fd1ff4a860529a80af61277f75c7e2/9/Dockerfile
+
+RUN groupadd --gid 1000 node \
+  && useradd --uid 1000 --gid node --shell /bin/bash --create-home node
+
+# gpg keys listed at https://github.com/nodejs/node#release-team
+RUN set -ex \
```
Comment (nylira) [inline tasks/build/Dockerfile]: This is cool, I like how we're verifying node and yarn versions through gpg keys.

**Comment by nylira** [comment]
I'm able to build and run the built Voyager apps with this PR (macOS and Windows).

**Comment by faboweb** [comment]
Nice work! 🚀 
Could you give an insight what the issue was?

File: tasks/build/Dockerfile (hunk 1 above)
Comment (NodeGuy) [reply tasks/build/Dockerfile]: Thanks, although I can't take credit for it.  That code is copied-and-pasted from the official Node.js Dockerfile.

FWIW I also question how valuable the exercise is given that it's not leveraging PGP's Web of Trust.

**Comment by NodeGuy** [comment]
I don't know exactly what the issue was.

I designed two phases into the build process: the first runs once and the results are cached to be re-used in further builds, and the second phase runs on every build.  When I originally designed the process we were using a long-lived version of Gaia so I put the downloading of Gaia into the first phase (and later replaced it with building the Cosmos SDK from source).  The issue appeared somewhere in there, including once on my machine, but I thought it was a fluke because I resolved it with `docker system prune`.

When I saw it happening to other people I decided to move the building of the Cosmos SDK into the second phase so that it happens on every build.  This also makes more sense now because we're changing the version of the Cosmos SDK we're using more rapidly than we used to.

**Comment by faboweb** [comment]
Ah nice. Thanks for the update.

**Pull request merged**